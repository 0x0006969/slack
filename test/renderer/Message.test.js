const { Message } = require('./../../lib/slack/renderer');

const issuesOpened = require('./../fixtures/webhooks/issues.opened.json');

describe('Message rendering', () => {
  test('works for messages with footers', async () => {
    const message = new Message({
      includeFooter: true,
      footerURL: issuesOpened.issue.html_url,
    });
    expect(message.getBaseMessage()).toMatchSnapshot();
  });

  test('works for messages without', async () => {
    const message = new Message({
      includeFooter: false,
    });
    expect(message.getBaseMessage()).toEqual({});
  });
});
