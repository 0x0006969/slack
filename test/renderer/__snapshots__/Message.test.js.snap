// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message rendering works for messages with footers 1`] = `
Object {
  "footer": "<https://github.com/github-slack/public-test/issues/1|View it on GitHub>",
  "footer_icon": "https://assets-cdn.github.com/favicon.ico",
}
`;
